server/src/item.rs

```diff
@@ -32,8 +32,32 @@ impl Item {
     }
 
     #[zbus(out_args("secret"))]
-    pub async fn get_secret(&self, _session: ObjectPath<'_>) -> Result<SecretInner, ServiceError> {
-        todo!()
+    pub async fn get_secret(
+        &self,
+        session: OwnedObjectPath,
+    ) -> Result<(SecretInner,), ServiceError> {
+        // wip
+        let inner = self.inner.lock().await;
+        let manager = self.manager.lock().await;
+        let secret = inner.secret();
+        let session = manager.session(&session);
+        if session.is_none() {
+            tracing::error!("The session does not exist");
+            return Err(ServiceError::NoSession("NoSession".to_string()));
+        }
+
+        let session = session.unwrap();
```

Comment:
Use let Some(session) = manager.session(&session) else { // the error bits } and you won't need the unwrap here.

Reply:
Done, thanks!